src/screens/feed/Feed.tsx

```diff
@@ -1,32 +1,108 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
 import { useNavigation } from '@react-navigation/native'
 import { LoadingModal } from 'components/LoadingModal'
 import { SafeAreaWrapper } from 'components/SafeAreaWrapper'
+import { useDeleteComment, useEditComment } from 'dataAccess/mutations/useAddReactionsComment'
 import { useGetPostsData } from 'dataAccess/queries/useFeedPostsData'
+import { useBooleanState } from 'hooks/useBooleanState'
 import { useLanguage } from 'hooks/useLanguage'
+import { EditTargetType } from 'mock-api/models/miragePostTypes'
 import { BottomTabNavigationProps } from 'navigation/types'
-import React, { useCallback, useEffect, useRef } from 'react'
 import { FlatList } from 'react-native'
-import { FeedPost as FeedPostType } from 'mockApi/models/miragePostTypes'
+import { OptionsModal } from 'components/OptionsModal'
+import EditIcon from 'assets/icons/icon-edit2.svg'
+import BinIcon from 'assets/icons/icon-bin.svg'
+import { MessageInputModal } from 'components/MessageInputModal'
+import { useNotifications } from 'react-native-notificated'
+import { useUserContext } from 'hooks/context-hooks/useUserContext'
 import { FeedHeader } from './components/FeedHeader/FeedHeader'
 import { FeedPost } from './components/FeedPost/FeedPost'
 
 const MAX_SCROLL_RETRIES = 4
 
 export const Feed = ({ route: { params: p } }: BottomTabNavigationProps<'FEED'>) => {
   const [language] = useLanguage()
+  const { notify } = useNotifications()
   const { data } = useGetPostsData()
   const navigation = useNavigation()
-
+  const { t } = useTranslation('feed')
+  const { user } = useUserContext()
   const flatListRef = useRef<FlatList | null>(null)
   const scrollRetries = useRef(0)
+
+  const [messageInputOpened, { setTrue: showMessageInput, setFalse: hideMessageInput }] =
+    useBooleanState(false)
+  const [isModalOpen, { setFalse: closeModal, setTrue: openModal }] = useBooleanState(false)
+  const [isEditingComment, { setFalse: setEditingCommentFalse, setTrue: setEditingCommentTrue }] =
+    useBooleanState(false)
```

Comment:
I think you may not need `isEditingComment` state as you can check

`if(editTarget.type === "comment")`